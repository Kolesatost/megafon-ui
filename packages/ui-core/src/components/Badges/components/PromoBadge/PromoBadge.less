@import (reference) '~styles/base.less';

@b: e('.mfui-promo-badge');

@{b} {
    display: flex;
    align-items: center;
    padding: 4px 12px;
    width: fit-content;

    border-radius: 8px;

    color: var(--stcWhite);

    &_type_hit {
        background-color: var(--warmRedC);
    }

    &_type_new {
        background-color: var(--systemBlue);
    }

    &_type_vip {
        background-color: var(--brandPurple);
    }

    &_type_popular {
        background-color: var(--137C);
    }

    &_type_user-choice {
        background-color: var(--reflexBlue);
    }

    &__text {
        .fontFamily();
        .smallFont();

        font-weight: 500;
    }
}
