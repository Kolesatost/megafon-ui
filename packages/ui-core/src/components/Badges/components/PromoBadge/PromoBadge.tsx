import * as React from 'react';
import { cnCreate, filterDataAttrs } from '@megafon/ui-helpers';
import * as PropTypes from 'prop-types';
import './PromoBadge.less';

export const PromoBadgeTypes = {
    HIT: 'hit',
    NEW: 'new',
    VIP: 'vip',
    POPULAR: 'popular',
    USER_CHOICE: 'user-choice',
} as const;

type PromoBadgeTypesType = typeof PromoBadgeTypes[keyof typeof PromoBadgeTypes];

export interface IPromoBadgeProps {
    /** Тип промо-бэйджа */
    type?: PromoBadgeTypesType;
    /** Дополнительный класс корневого элемента */
    className?: string;
    /** Дополнительные data-атрибуты к внутренним элементам */
    dataAttrs?: {
        root?: Record<string, string>;
    };
    children: JSX.Element[] | Element[] | JSX.Element | Element | string;
}

const cn = cnCreate('mfui-promo-badge');
const PromoBadge: React.FC<IPromoBadgeProps> = ({ type = 'hit', className, children, dataAttrs }) => (
    <div {...filterDataAttrs(dataAttrs?.root)} className={cn({ type }, className)}>
        <span className={cn('text')}>{children}</span>
    </div>
);

PromoBadge.propTypes = {
    type: PropTypes.oneOf(Object.values(PromoBadgeTypes)),
    className: PropTypes.string,
    dataAttrs: PropTypes.shape({
        root: PropTypes.objectOf(PropTypes.string.isRequired),
    }),
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.element.isRequired),
        PropTypes.element,
        PropTypes.string,
    ]).isRequired,
};

export default PromoBadge;
