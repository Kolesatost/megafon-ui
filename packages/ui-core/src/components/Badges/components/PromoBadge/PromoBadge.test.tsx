import * as React from 'react';
import { shallow } from 'enzyme';
import PromoBadge from './PromoBadge';

const defaultChildren = 'badge text';

describe('<PromoBadge />', () => {
    describe('snapshots', () => {
        it('renders with default props', () => {
            const wrapper = shallow(<PromoBadge>{defaultChildren}</PromoBadge>);

            expect(wrapper).toMatchSnapshot();
        });

        it('renders with type prop', () => {
            const wrapper = shallow(<PromoBadge type="popular">{defaultChildren}</PromoBadge>);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
