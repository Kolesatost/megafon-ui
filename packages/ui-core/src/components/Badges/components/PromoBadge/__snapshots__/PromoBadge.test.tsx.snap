// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PromoBadge /> snapshots renders with default props 1`] = `
<div
  className="mfui-promo-badge mfui-promo-badge_type_hit"
>
  <span
    className="mfui-promo-badge__text"
  >
    badge text
  </span>
</div>
`;

exports[`<PromoBadge /> snapshots renders with type prop 1`] = `
<div
  className="mfui-promo-badge mfui-promo-badge_type_popular"
>
  <span
    className="mfui-promo-badge__text"
  >
    badge text
  </span>
</div>
`;
